crypto/cms/cms_rsa.c

```diff
@@ -147,20 +148,21 @@ static int rsa_cms_encrypt(CMS_RecipientInfo *ri)
     if (!ossl_x509_algor_md_to_mgf1(&oaep->maskGenFunc, mgf1md))
         goto err;
     if (labellen > 0) {
-        ASN1_OCTET_STRING *los = ASN1_OCTET_STRING_new();
+        los = ASN1_OCTET_STRING_new();
 
         if (los == NULL)
             goto err;
-        if (!ASN1_OCTET_STRING_set(los, label, labellen)) {
-            ASN1_OCTET_STRING_free(los);
+        if (!ASN1_OCTET_STRING_set(los, label, labellen))
             goto err;
-        }
+
         oaep->pSourceFunc = ossl_X509_ALGOR_from_nid(NID_pSpecified,
                                                      V_ASN1_OCTET_STRING, los);
-        if (oaep->pSourceFunc == NULL)
+        if (!oaep->pSourceFunc)
```

Comment:
Please revert this.  Pointers should be always compared with NULL.

Reply:
done.